### PR TITLE
enable stream-style for mgmt, prepare release 4.1.31

### PR DESCRIPTION
**Pull request by XiaofeiCao**

### CODE REVIEW

**Review by weidongxu-microsoft — APPROVED** [review]
You probably need to exclude some fluent CI from azure-json...

**Comment by XiaofeiCao** [comment]
> You probably need to exclude some fluent CI from azure-json...

Yeah, checking which one failed.

File: fluent-tests/Initialize-Tests.ps1 [hunk 1]
```diff
@@ -91,7 +91,7 @@ $job = @(
     "--version=$AUTOREST_CORE_VERSION $FLUENT_ARGUMENTS --input-file=https://raw.githubusercontent.com/Azure/azure-rest-api-specs/main/specification/cdn/resource-manager/Microsoft.Cdn/stable/2020-09-01/afdx.json --namespace=com.azure.mgmttest.afdx",
 
     # nested x-ms-flatten from superclass in ExtendedProduct
-    "--version=$AUTOREST_CORE_VERSION $FLUENT_ARGUMENTS --input-file=https://raw.githubusercontent.com/Azure/azure-rest-api-specs/main/specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/Product.json --namespace=com.azure.mgmttest.azurestack",
+    "--version=$AUTOREST_CORE_VERSION $FLUENT_ARGUMENTS --input-file=https://raw.githubusercontent.com/Azure/azure-rest-api-specs/main/specification/azurestack/resource-manager/Microsoft.AzureStack/preview/2020-06-01-preview/Product.json --namespace=com.azure.mgmttest.azurestack --stream-style-serialization=false",
```
Comment (XiaofeiCao) [inline fluent-tests/Initialize-Tests.ps1:94]: azurestack is in the fluentgen report error: https://github.com/Azure/autorest.java/issues/2823

File: fluent-tests/Initialize-Tests.ps1 [hunk 2]
```diff
@@ -100,7 +100,7 @@ $job = @(
     "--version=$AUTOREST_CORE_VERSION $FLUENT_ARGUMENTS --input-file=https://raw.githubusercontent.com/Azure/azure-rest-api-specs/main/specification/hybridnetwork/resource-manager/Microsoft.HybridNetwork/stable/2021-05-01/device.json --namespace=com.azure.mgmttest.hybridnetwork",
 
     # flatten the empty model which has non-empty parent model
-    "--version=$AUTOREST_CORE_VERSION $FLUENT_ARGUMENTS --input-file=https://github.com/Azure/azure-rest-api-specs/blob/main/specification/monitor/resource-manager/Microsoft.Insights/preview/2021-09-01-preview/dataCollectionRules_API.json --namespace=com.azure.mgmttest.monitor",
+    "--version=$AUTOREST_CORE_VERSION $FLUENT_ARGUMENTS --input-file=https://github.com/Azure/azure-rest-api-specs/blob/main/specification/monitor/resource-manager/Microsoft.Insights/preview/2021-09-01-preview/dataCollectionRules_API.json --namespace=com.azure.mgmttest.monitor --stream-style-serialization=false",
```
Comment (XiaofeiCao) [inline fluent-tests/Initialize-Tests.ps1:103]: monitor is also in the error report

File: fluent-tests/src/test/java/com/azure/mgmttest/RuntimeTests.java [hunk 3]
```diff
@@ -117,6 +117,7 @@ public void testManagementClient() {
     }
 
     @Test
+    @Disabled("Enable when deserialization for error subclasses are fixed.")
```
Comment (XiaofeiCao) [inline fluent-tests/src/test/java/com/azure/mgmttest/RuntimeTests.java:120]: Stream-style for error subclass seems to have issue. 
Created: https://github.com/Azure/autorest.java/issues/2826

Comment (weidongxu-microsoft) [reply fluent-tests/src/test/java/com/azure/mgmttest/RuntimeTests.java:120]: Just use `steam-style-serialization=false` for this case. Don't disable the test.

File: package.json
Comment (weidongxu-microsoft) [inline package.json:1]: package-lock

File: fluent-tests/src/test/java/com/azure/mgmttest/RuntimeTests.java (hunk 3 above)
Comment (XiaofeiCao) [reply fluent-tests/src/test/java/com/azure/mgmttest/RuntimeTests.java]: updated

File: package.json
Comment (XiaofeiCao) [reply package.json:1]: updated

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**